README.md

```diff
@@ -1,65 +1,131 @@
+It's recommended that you use [`multirust`](https://github.com/brson/multirust)
+to install and switch between versions of Rust.
 
-## Cross Compilation Toolchain
+### Node
 
-If you prefer to cross-compile on your dev system, this is the way to go. There is an extensive write-up of the process at https://github.com/japaric/rust-cross. There's also an experimental, pre-compiled toolchain at 
-https://people.mozilla.org/~fdesre/rust-rpi2.tar.gz which Fabrice built on Ubuntu 15.10. It is linked against a specific version of libstdc++.so.6 (GLIBCXX_3.4.21), so it may or may not work on other distributions. Ubuntu 15.04 is reportedly no good.
+We're using Node to run Selenium tests. Currently v4.x LTS. We plan to stay on
+stable LTS releases. It's recommended that you use
+[`nvm`](https://github.com/creationix/nvm) to install and switch between
+versions of Node.
 
- * Download and and unpack toolchain linked above in $toolchain
- * Add $toolchain/x-tools/bin and $toolchain/bin to your PATH
- * Add $toolchain/lib to your LD_LIBRARY_PATH
 
-To build a rust file:
+## Target hardware
 
-``` bash
-$ rustc --target=armv7-unknown-linux-gnueabihf -C linker=armv7-unknown-linux-gnueabihf-g++ hello.rs
-```
+We're using the Raspberry Pi 2 as a prototyping target (ARMv7). The target
+operating system is the latest Raspbian which is based on Debian 8.0 Jessie.
 
-## Building and running locally on Linux
 
-This should work straight-forward. Install a rust nightly via multirust, clone the repo, and then cargo run.
+## Contributing
 
+Note: We're in an iterative prototyping phase of the project. Things are moving
+really fast so it may be easier to contribute when the dust starts to settle.
+You've been warned. :shipit:
 
-## Building and running locally on OS X
+### Forks and feature branches
 
-``` bash
-$ brew install openssl ... ... ... # TODO: other build dependencies?
-$ brew install multirust
-$ multirust update
-$ multirust default nightly
+You should fork the main repo and create pull requests against feature branches
+of your fork. If you need some guidance with this see:
+
+ - https://guides.github.com/introduction/flow/
+ - http://scottchacon.com/2011/08/31/github-flow.html
+
+
+## Setup
+
+```bash
+$ git clone git@github.com:<username>/foxbox.git
+$ cd foxbox
 ```
 
-This is required to build the openssl crate using homebrew's openssl:
+## Running the deamon
 
-``` bash
-$ export DEP_OPENSSL_INCLUDE=/usr/local/opt/openssl/include/
-$ export OPENSSL_INCLUDE_DIR=/usr/local/Cellar/openssl/1.0.2f/include/
+```bash
+$ cargo run
 ```
 
-This then builds and runs the project locally:
+Alternativly you can build the app without running it via:
 
-``` bash
-$ git clone https://github.com/fxbox/foxbox
-$ cd foxbox
+```bash
 $ cargo build
-$ cargo run
 ```
 
 
-## Building on Raspbian
+## Rust tests
+
+```bash
+$ cargo test
+```
+
+
+## Selenium tests
+
+You'll need to make sure you install the dependencies via:
+
+```bash
+$ npm install
+```
```

Comment:
You'll also need to install nodejs, and [install and start the selenium server](https://github.com/fxbox/foxbox/blob/49e94cd2446377e5952f61580e7303b6a66531af/.travis.yml#L10-L13) (see also https://github.com/fxbox/foxbox/pull/19 - maybe we can install and start selenium from package.json).